src/main/java/org/mars/rover/kata/MarsNavigator.java

```diff
@@ -39,21 +44,26 @@ protected void createGrid() {
         }
     }
 
-    public void constructRovers() {
-        providedInput.roverInstructions().forEach((RoverInstructions instructions) -> {
-            var initialPosition = instructions.initialPosition();
+    public List<MarsRover> constructRovers(List<RoverInstructions> roverInstructions) {
+        return roverInstructions.stream()
+                .map(RoverInstructions::initialPosition)
+                .map(initialPosition -> new MarsRover(
+                        initialPosition.x(),
+                        initialPosition.y(),
+                        initialPosition.direction()
+                ))
+                .toList();
+    }
 
-            MarsRover marsRover = new MarsRover(
-                    initialPosition.x(),
-                    initialPosition.y(),
-                    initialPosition.direction()
-            );
+    public static MarsNavigator fromString(String inputString) {
+        var commandSet = new StdinProcessor(new CommandParser()).processInput(new Scanner(
```

Comment:
To wygląda podejrzanie skomplikowanie 😉 Czy np. `CommandParser` nie mógłby po prostu być polem `StdinProcessor`a, jako szczegół implementacyjny?

Podobnie `Scanner` wygląda na pierwszy rzut oka jak szczegół implementacyjny funkcji `processInput`

Reply:
Dodane.